### ISSUE TITLE
Cookie headers prefixed with $ are parsed incorrectly

### ISSUE THREAD

**Issue by jonluca** [curlconverter/curlconverter#207]
Chrome will let you export a network call as curl by right clicking it -> copy -> copy as curl. 

Chrome also likes to export long cookies like such:

```
curl 'https://someurl.com' \
  -H 'Connection: keep-alive' \
  -H 'Cache-Control: max-age=0' \
  -H 'sec-ch-ua: "Google Chrome"; v="83"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'DNT: 1' \
  -H 'Upgrade-Insecure-Requests: 1' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4093.3 Safari/537.36' \
  -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9' \
  -H 'Sec-Fetch-Site: none' \
  -H 'Sec-Fetch-Mode: navigate' \
  -H 'Sec-Fetch-User: ?1' \
  -H 'Sec-Fetch-Dest: document' \
  -H 'Accept-Language: en-US,en;q=0.9,it;q=0.8,la;q=0.7' \
  -H $'Cookie: somereallyreallylongcookie=true;' \
  --compressed
```

This gets parsed as 

```py
cookies = {
    '$Cookie: somereallyreallylongcookie': 'true',
}
```

in python, while in reality the $ should effectively be omitted, and the cookie itself should be parsed correctly.

**Comment by mlissner** [comment]
This is affecting multi-part form data as well. From copy-as-curl in Firefox, I got the following (slightly modified command):

    curl 'http://localhost:3001' -H 'Content-Type: multipart/form-data; boundary=---------------------------22070739573223388867407661007' \
        --data-binary $'-----------------------------22070739573223388867407661007\r\nContent-Disposition: form-data; name="caseid"\r\n\r\n165646\r\n-----------------------------22070739573223388867407661007--\r\n'

Sending that to `nc -l 3001`, I get the correct response of:

```
POST / HTTP/1.1
Host: localhost:3001
User-Agent: curl/7.65.3
Accept: */*
Content-Type: multipart/form-data; boundary=---------------------------22070739573223388867407661007
Content-Length: 179

-----------------------------22070739573223388867407661007
Content-Disposition: form-data; name="caseid"

165646
-----------------------------22070739573223388867407661007--
```

When I convert the curl command from Firefox into a Python requests code via the converter, it gives me:

```
import requests

headers = {
    'Content-Type': 'multipart/form-data; boundary=---------------------------22070739573223388867407661007',
}

data = '$-----------------------------22070739573223388867407661007\\r\\nContent-Disposition: form-data; name="caseid"\\r\\n\\r\\n165646\\r\\n-----------------------------22070739573223388867407661007--\\r\\n'

response = requests.post('http://localhost:3001/', headers=headers, data=data)
```

Running against `nc -l 3001` gives me:

```
POST / HTTP/1.1
Host: localhost:3001
Connection: keep-alive
Accept-Encoding: gzip, deflate
Accept: */*
User-Agent: python-requests/2.20.0
Content-Type: multipart/form-data; boundary=---------------------------22070739573223388867407661007
Content-Length: 190

$-----------------------------22070739573223388867407661007\r\nContent-Disposition: form-data; name="caseid"\r\n\r\n165646\r\n-----------------------------22070739573223388867407661007--\r\n
```

Which, alas, is very much not the same thing as the `nc` response above. Sigh.

But if I remove the dollar sign from curl:

```
curl 'http://localhost:3001' -H 'Content-Type: multipart/form-data; boundary=---------------------------22070739573223388867407661007' --data-binary '-----------------------------22070739573223388867407661007\r\nContent-Disposition: form-data; name="caseid"\r\n\r\n165646\r\n-----------------------------22070739573223388867407661007--\r\n'
```

Then curl gives me the same (wrong) thing that the generated `requests` command did:

```
POST / HTTP/1.1
Host: localhost:3001
User-Agent: curl/7.65.3
Accept: */*
Content-Type: multipart/form-data; boundary=---------------------------22070739573223388867407661007
Content-Length: 189

-----------------------------22070739573223388867407661007\r\nContent-Disposition: form-data; name="caseid"\r\n\r\n165646\r\n-----------------------------22070739573223388867407661007--\r\n
```

I honestly don't know bash well enough to understand what the dollar sign is doing here, but Firefox's copy-as-curl command put it there, so I imagine it's not *so* uncommon? Seems like the parser doesn't know what to do with it and that it has some special meaning.

**Comment by mlissner** [comment]
Ah ha. `$'...'` has a special meaning in bash, which is to actually interpret escape sequences (like `\n\r`). Makes sense: https://stackoverflow.com/a/11966402/64911

**Comment by jonluca** [comment]
Nice, great investigation!

Should be pretty straight forward to have it unescape strings prefixed with `$'`

**Comment by kylezervas** [comment]
> Chrome will let you export a network call as curl by right-clicking it -> copy -> copy as curl.
> 
> Chrome also likes to export long cookies like such:
> 
> ```
> curl 'https://someurl.com' \
>   -H 'Connection: keep-alive' \
>   -H 'Cache-Control: max-age=0' \
>   -H 'sec-ch-ua: "Google Chrome"; v="83"' \
>   -H 'sec-ch-ua-mobile: ?0' \
>   -H 'DNT: 1' \
>   -H 'Upgrade-Insecure-Requests: 1' \
>   -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4093.3 Safari/537.36' \
>   -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9' \
>   -H 'Sec-Fetch-Site: none' \
>   -H 'Sec-Fetch-Mode: navigate' \
>   -H 'Sec-Fetch-User: ?1' \
>   -H 'Sec-Fetch-Dest: document' \
>   -H 'Accept-Language: en-US,en;q=0.9,it;q=0.8,la;q=0.7' \
>   -H $'Cookie: somereallyreallylongcookie=true;' \
>   --compressed
> ```
> 
> This gets parsed as
> 
> ```python
> cookies = {
>     '$Cookie: somereallyreallylongcookie': 'true',
> }
> ```
> 
> in python, while in reality the $ should effectively be omitted, and the cookie itself should be parsed correctly.

**Comment by kylezervas** [comment]
Duplicate of #

**Comment by jonluca** [comment]
I put up a PR to fix this in yargs https://github.com/yargs/yargs-parser/pull/365

**Comment by NickCarneiro** [comment]
This is excellent. Thanks for going above and beyond, @jonluca

**Comment by verhovsky** [comment]
This was solved by #278
